### ISSUE TITLE
Serialization/Deserialization in .net core 5.0

### ISSUE THREAD

**Issue by jstachera** [LordMike/TMDbLib#360]
There is problem during Json deserialization for list of TvShows which were previously serialized in .net core 5.0 using serializer from System.Text.Json:

`var jsonStr = JsonSerializer.Serialize(obj); // obj is List<TvShow>`

I got error message:

> Newtonsoft.Json.JsonSerializationException: 'Could not create an instance of type TMDbLib.Objects.Changes.ChangeItemBase. Type is an interface or abstract class and cannot be instantiated. Path '[19].Changes.Changes[0].Items[0].Action', line 1, position 434211.'

**Comment by LordMike** [comment]
Yea, we use newtonsoft converters to handle some special cases, like the one you mention where the base type (an abstract) has multiple derived types. Our converter picks the correct type at runtime.

To support system.text.json, we need to do the same.

**Comment by jstachera** [comment]
What do you mean 'our converter'? Is it provided with the library? How can I use it?  do you have any sample of json serialization/dserialization ?

**Comment by LordMike** [comment]
Sure - it's here:

https://github.com/LordMike/TMDbLib/blob/8d61234d9f2094d0d1e3926cc784e9f419907a2a/TMDbLib/Client/TMDbClient.cs#L31-L37

All the converters are here:
https://github.com/LordMike/TMDbLib/tree/master/TMDbLib/Utilities/Converters

So yea - it's included and used in the client. 

I'm a bit curious though. Why would you get that exception when serializing stuff? .. I'd expect it if you were deserializing data, as this is where the concrete type needs to be identified.

**Comment by lewishenson** [comment]
Do you have any current plans to switch to using System.Text.Json classes?

**Comment by lewishenson** [comment]
I could have a little play. I've just grabbed the code and run the tests and unfortunately I've got failing tests so I'm not sure how confident I would be with my changes. Do I need to do anything special to run the tests?

**Comment by LordMike** [comment]
Some tests are failing right now, which is also (part of) why we don't run tests in CI (the other part being #300).

If you make changes and _most_ of the tests work, I'd say that's good enough. There are some "UtilityTests" which I can see are related to json conversion - they test the converters directly, so in reality these tests should be rewritten to use a "TMDbLib Converter" ... which also should be used for all serialization ..

_Then_ you could consistently make changes. 

So basically a small refactoring to pack the newtonsoft dependency into one class, and then replace that class.

**Comment by lewishenson** [comment]
Ah good to know thanks. I might take a look this weekend then.

**Comment by LordMike** [comment]
I've refactored the serialization into a new `ITMDbSerializer` interface. This change removes all Newtonsoft.Json from all API surfaces.. 

* The constructor will no longer take a JsonSerializer
* The Newtonsoft.Json dependency is hidden, so it is no longer transient
* There is a `TMDbJsonSerializer.Instance` which has the internal serializer
